WordPress/Tests/Theme/NoHardCodedUrlsUnitTest.inc

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+* Test file : NoHardCodedUrls
+* Theme URI: https://mysite.org/themes/mytheme/    // OK
+* Author URI: https://mysite.org/                        // OK
+* License URI: http://www.gnu.org/licenses/gpl-2.0.html     // OK
+*
+*/
+?>
+<link rel="profile" href="http://gmpg.org/xfn/11">  <!-- OK -->
+<link rel="pingback" href="<?php bloginfo( 'pingback_url' ); ?>">  <!-- OK -->
+<link href='http://fonts.googleapis.com/css?family=Lato' rel='stylesheet' type='text/css' />  <!-- OK but not really :) -->
+<script src="https://code.jquery.com/jquery-1.10.2.js"></script>  <!-- bad -->
+<!-- http://ajax.googleapis.com/ajax/libs/jquery/1.7.2/jquery.min.js --> <!-- OK url in comment tags -->  <!-- OK -->
+<?php
+wp_register_script( 'jquery', 'http://ajax.googleapis.com/ajax/libs/jquery/1.7.2/jquery.min.js');  // bad
+
+if ( $fonts ) {
+		$fonts_url = add_query_arg( array(
+			'family' => urlencode( implode( '|', $fonts ) ),
+			'subset' => urlencode( $subsets ),
+		), 'https://fonts.googleapis.com/css' );  // OK
+}
+?>
+<a href="<?php echo esc_url( __( 'https://wordpress.org/', 'twentysixteen' ) ); ?>"> <!-- OK -->
+  <?php printf( __( 'Proudly powered by %s', 'twentysixteen' ), 'WordPress' ); ?></a>
+  <a href="<?php echo esc_url( __( 'https://wordpress.com/', 'twentysixteen' ) ); ?>"> <!-- OK -->
+    <?php printf( __( 'Proudly powered by %s', 'twentysixteen' ), 'WordPress' ); ?></a>
+<?php
+$foo="https://mysite.wow.com/"."hello"; //bad
+// Comment test http://youtube.com   //OK
+/////// Special comment test //mysite.com ////// //OK
+$http_prefix1 = "http://"; // OK
+$http_prefix2 = 'https://'; // OK
+$http_prefix3 = "//";  // OK
+$http_prefix1 = "http:"; // OK
+$http_prefix2 = 'https:'; // OK
+//hello.org // OK
+_e( 'Theme by <a href="//mizmizi.com/">Mizmizi</a> Powered by <a href="http://wordpress.org/">WordPress ', 'tdomain' ); // Bad
```

Comment:
You may want to add some variations of the [example I gave above](https://github.com/WPTRT/WordPress-Coding-Standards/pull/28/files#r70921284) to the test file.